### ISSUE TITLE
Portal editor validation is not working

### ISSUE THREAD

**Issue by laurenwalker** [NCEAS/metacatui#1850]
I  am getting a save error in the portal editor in  the `feature-rule-groups` branch when I try to create a  new portal but don't create  a collection query in the  query builder.

The expectation is that the  editor should catch this  missing piece in the validation process OR a default collection query should be created instead.

## Steps to reproduce
1. Open up a fresh portal editor at /edit/portals
2. Fill out a label, title, and description
3. Click save
4. See the save error

![Screen Shot 2021-08-19 at 10.59.08 AM.png](https://images.zenhubusercontent.com/5d152e12c27d0c306f021863/fe74ee34-87de-4cf1-a81c-f2579d21f0e6)

**Comment by robyngit** [comment]
This was fixed with commit 24313c56604a8808b9073a9ce0f82345ebe48dc2. It was the same bug that was causing issue #1851. I confirmed that a new portal could be saved successfully without adding any query rules -  this is allowed because a hidden "isPartOf" filter is added to the definition in new portals. If the isPartOf filter is visible (set `hideIsPartOfFilter: false` in the `AppModel`), and it's removed before saving the new portal, then the empty definition filter is flagged appropriately:

<img width="1134" alt="Screen Shot 2021-09-10 at 17 14 42" src="https://user-images.githubusercontent.com/26600641/132918487-031e7bf7-9e62-4583-9151-570276dfe6e1.png">
